### PR TITLE
Added open magic wallet functionality 

### PR DESCRIPTION
**Pull request by Israellund**
## Link to Issue
Closes: #5309 

## Description of Changes
- Added open magic wallet functionality for accounts made through email and Google

**NOTE: I made these changes based off of the [Figma design](https://www.figma.com/design/rBTow8pp1mvKyWmEM8z3aP/Magic-Wallet?node-id=2-1789&node-type=frame&t=07MJEJ7PgtSBjCs4-0)s. I tried desperately to embed the widget in the account creation modal as was shown in the original designs, but unfortunately widget's don't work that way. Sabina and I decided to simply replace it with an image and change the copy telling the user where they can access their magic wallet**

## Test Plan
- Sign in with email
- confirm you see the arrow-square-out icon, and when you hover over it a tooltip reads "Open wallet"
- click the icon and confirm you see the Magic wallet widget
- go to a community with Stake and click "Buy Stake"
- confirm you see the "Add funds" button
- click it and confirm you see the Magic widget
- click the profile dropdown and confirm you see "Open wallet" with the arrow-square-out icon
- click either the text or the icon and confirm that you see the Magic widget
- in the Profile dropdown click "My community stake" 
- on My Community Stake page confirm you see the Open wallet button  
- click the button and confirm you see the Magic widget.
- Sign in with Google and go through all of the above steps
- Sign in with any other wallet/method and confirm that you do not see any of the above, as they are set to only show when a user has a Magic wallet. 
Screenshot 2024-11-25 at 10 12 43 AM
Screenshot 2024-11-25 at 10 13 00 AM
Screenshot 2024-11-25 at 10 13 10 AM
Screenshot 2024-11-25 at 10 13 20 AM
Screenshot 2024-11-25 at 10 14 14 AM

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/modals/ManageCommunityStakeModal/StakeExchangeForm/StakeExchangeForm.tsx [hunk 1]
```diff
@@ -129,6 +133,9 @@ const StakeExchangeForm = ({
 
   const { isAddedToHomeScreen } = useAppStatus();
 
+  const userData = useUserStore();
+  const hasMagic = userData.addresses?.[0]?.walletId === WalletId.Magic;
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/modals/ManageCommunityStakeModal/StakeExchangeForm/StakeExchangeForm.tsx:137]: same as above

File: packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/WelcomeOnboardModal.tsx [hunk 2]
```diff
@@ -60,6 +61,19 @@ const WelcomeOnboardModal = ({ isOpen, onClose }: WelcomeOnboardModalProps) => {
...
+            />
+          ),
+        };
+      }
+      case WelcomeOnboardModalSteps.MagicWallet: {
+        return {
+          index: 4,
+          title: 'Magic Wallet Creation',
+          component: (
+            <MagicWalletCreationStep
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/WelcomeOnboardModal.tsx:76]: magic wallets are only made for sms + other sso methods right? If yes shouldn't this be displayed conditionally when a signup with one of those method happens?

File: libs/shared/src/types/protocol.ts [hunk 3]
```diff
@@ -83,6 +83,8 @@ export enum WalletSsoSource {
   Twitter = 'twitter',
   Apple = 'apple',
   Email = 'email',
+  Farcaster = 'farcaster',
```
Comment (mzparacha) [inline libs/shared/src/types/protocol.ts:86]: is farcaster logic a part of this pr?

File: packages/commonwealth/client/scripts/views/components/SublayoutHeader/useUserMenuItems.tsx [hunk 4]
```diff
@@ -163,6 +165,14 @@ const useUserMenuItems = ({
...
   }, [updateCanvasSignedAddresses]);
 
+  const openMagicWallet = async () => {
+    try {
+      await magic.wallet.showUI();
+    } catch (error) {
+      console.trace(error);
+    }
+  };
+
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/SublayoutHeader/useUserMenuItems.tsx:175]: move to useAuthentication

**Review by mzparacha — APPROVED** [review]
SMS login works, didn't test other sso methods. Codewise lgtm

**Review by ianrowan — APPROVED** [review]
tested, LGTM from my POV other than Malik's comments + CI failure

File: libs/shared/src/types/protocol.ts (hunk 3 above)
Comment (Israellund) [reply libs/shared/src/types/protocol.ts:86]: @mzparacha this branch was made off of a branch Kurtis made, at the request of Dillon. So I believe that is a change he made.

File: packages/commonwealth/client/scripts/views/pages/AdminPanel/UpdateCustomDomainTask.tsx [hunk 5]
```diff
@@ -25,6 +25,8 @@ const UpdateCustomDomainTask = () => {
           buttonType: 'destructive',
           buttonHeight: 'sm',
           onClick: () => {
+            console.log('communityId', communityId);
+            console.log('customDomain', customDomain);
```
[the remark this answers is not in the document]
Comment (Israellund) [reply packages/commonwealth/client/scripts/views/pages/AdminPanel/UpdateCustomDomainTask.tsx:29]: lol nope

**Pull request merged**